docs/source/examples/vqe-pyquil-demo.myst

```diff
@@ -0,0 +1,312 @@
+---
+jupytext:
+  text_representation:
+    extension: .myst
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.10.3
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Variational Quantum Eigensolver improved with Zero Noise Extrapolation
+
+In this example we investigate how Zero Noise Extrapolation (ZNE) can improve 
+convergence when applied to a variational problem. ZNE works by computing the 
+observable of interest at increased noise levels, i.e. beyond the minimum noise 
+strength in the computer, and then extrapolating back to the zero-noise limit. 
+The two main components of ZNE are noise scaling and extrapolation. Read more about 
+ZNE in the [[Mitiq User Guide]]
+(https://mitiq.readthedocs.io/en/stable/guide/guide-zne.html#noise-scaling-by-unitary-folding).\
+\
+
+The Variational Quantum Eigensolver (VQE) is a hybrid quantum-classical algorithm used to
+solve eigenvalue and optimization problems. The VQE algorithm consists of a quantum 
+subroutine run inside of a classical optimization loop. In this example, the goal of the 
+optimization is to find the smallest eigenvalue of a matix H, which is the Hamiltonian 
+of a simple quantum system. The quantum subroutine prepares the quantum state 
+|Ψ(vec(θ))⟩ and measures the expectation value ⟨Ψ(vec(θ))|H|Ψ(vec(θ))⟩. By the 
+variational principle, ⟨Ψ(vec(θ))|H|Ψ(vec(θ))⟩ is always greater than the smallest 
+eigenvalue of H, which means a classical optimization loop can be used to find this 
+eigenvalue. \
+\
+
+The VQE example shown here is adapted from the `VQE` function in Grove 
+[[1]](#References) and the pyQuil / Grove VQE tutorial [[2]](#References). 
+
+
+## Defining the quantum system using pyQuil
+
+```{code-cell} ipython3
+import numpy as np
+from pyquil import get_qc, Program
+from pyquil.gates import RX, RY, S, T, Z, CNOT, MEASURE
+from pyquil.paulis import PauliTerm, PauliSum, sZ
+from pyquil.noise import pauli_kraus_map, append_kraus_to_gate
+from typing import List, Union
+from collections import Counter
+from matplotlib import pyplot as plt
+from scipy import optimize
+import mitiq
+from mitiq import zne
+from mitiq.zne.scaling.folding import fold_gates_from_right
+```
+
+Use the `get_qc` command to initialize the simulated backend 
+where the pyQuil program will run
+
+```{code-cell} ipython3
+backend = get_qc("2q-qvm")
+```
+
+Define example ansatz, consisting of a rotation by angle `theta` and a layer of static gates: 
+
+```{code-cell} ipython3
+program = Program()
+theta = program.declare("theta", memory_type="REAL")
+program += RX(theta, 0)
+program += T(0)
+program += CNOT(1, 0)
+program += S(0)
+program += Z(0)
```

Comment:
This ansatz is a little strange  - the only gate that does anything (without noise) is RX(theta, 0).

Reply:
I know...I used it to add noise via static gates, and I wanted the ideal energy expectation to have a clear minimum.